### PR TITLE
Improve selecting non media attachments

### PR DESCRIPTION
**Pull request by adasiewiczr**
# Submit a pull request

## CLA

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required).
- [x] The code changes follow best practices
- [x] Code changes are tested (add some information if not applicable)

## Description of the pull request
This PR replaces importing files (using ACTION_OPEN_DOCUMENT_TREE) when selecting non-media attachments with showing system dialog (using ACTION_OPEN_DOCUMENT). Moreover, selecting files using ACTION_OPEN_DOCUMENT doesn't require READ_EXTERNAL_STORAGE permission so the file permission indicator was removed.
It also fixes showing attachments captured with the camera.
The cleanup (unused methods and classes) will be removed in the next PR

### CODE REVIEW

File: library/src/main/java/com/getstream/sdk/chat/model/ModelType.kt [hunk 1]
```diff
@@ -48,4 +49,18 @@ object ModelType {
     const val action_send = "send"
     const val action_shuffle = "shuffle"
     const val action_cancel = "cancel"
+
+    val supportedMimeTypes = listOf(
```
Comment (ogkuzmin) [inline library/src/main/java/com/getstream/sdk/chat/model/ModelType.kt:53]: I see it duplicates `StorageHelper::supportedFilesMimeTypes`. I'd like to keep only one

File: library/src/main/java/com/getstream/sdk/chat/utils/StorageHelper.kt [hunk 2]
```diff
@@ -89,6 +89,28 @@ internal class StorageHelper {
...
+                    type = ModelType.attach_file,
+                    mimeType = context.contentResolver.getType(uri),
+                    title = cursor.getString(cursor.getColumnIndex(OpenableColumns.DISPLAY_NAME))
+                ).apply {
+                    this.size = cursor.getLong(cursor.getColumnIndex(OpenableColumns.SIZE))
+                }
+            }
+        }
+
+        return attachments
```
Comment (ogkuzmin) [inline library/src/main/java/com/getstream/sdk/chat/utils/StorageHelper.kt:111]: We can omit a mutable collection

```suggestion
        return uriList.mapNotNull { uri ->
            context.contentResolver.query(uri, null, null, null, null)?.use { cursor ->
                cursor.moveToFirst()
                AttachmentMetaData(
                    uri = uri,
                    type = ModelType.attach_file,
                    mimeType = context.contentResolver.getType(uri),
                    title = cursor.getString(cursor.getColumnIndex(OpenableColumns.DISPLAY_NAME))
                ).apply {
                    this.size = cursor.getLong(cursor.getColumnIndex(OpenableColumns.SIZE))
                }
            }
        }
```

File: library/src/main/java/com/getstream/sdk/chat/view/messageinput/AttachmentsController.kt [hunk 3]
```diff
@@ -65,18 +65,17 @@ internal class AttachmentsController(
...
         }
     }
 
-    private fun filterMediaFiles(isMedia: Boolean, filesToFilter: Set<AttachmentMetaData>): Set<AttachmentMetaData> {
-        return if (!isMedia) {
-            filesToFilter
-        } else {
-            filesToFilter.filter(mediaAttachmentsPredicate).toSet()
-        }
+    private fun filterMediaFiles(
```
Comment (ogkuzmin) [inline library/src/main/java/com/getstream/sdk/chat/view/messageinput/AttachmentsController.kt:73]: Naming of this function doesn't relate its' logic anymore. Before it filtered only media files, now for media and non-media

File: library/src/main/java/com/getstream/sdk/chat/view/messageinput/MessageInputController.kt [hunk 4]
```diff
@@ -152,9 +169,20 @@ internal class MessageInputController(
     }
 
     internal fun onFileCaptured(file: File) {
+        attachmentsController.setSelectedAttachmentAdapter(null, true)
         attachmentsController.selectAttachment(AttachmentMetaData(file), true)
     }
 
+    internal fun onFilesSelected(uriList: List<Uri>) {
+        GlobalScope.launch(dispatchersProvider.mainDispatcher) {
```
Comment (ogkuzmin) [inline library/src/main/java/com/getstream/sdk/chat/view/messageinput/MessageInputController.kt:177]: For me it's logic related to work with attachments. That's why this should be moved to AttachmentsController

**Review by ogkuzmin — CHANGES_REQUESTED** [review]

File: library/src/main/java/com/getstream/sdk/chat/model/ModelType.kt (hunk 1 above)
Comment (adasiewiczr) [reply library/src/main/java/com/getstream/sdk/chat/model/ModelType.kt:53]: Agree, I've removed `StorageHelper::supportedFilesMimeTypes` in #677 (I didn't want to make too many changes in this PR)

File: library/src/main/java/com/getstream/sdk/chat/utils/StorageHelper.kt (hunk 2 above)
Comment (adasiewiczr) [reply library/src/main/java/com/getstream/sdk/chat/utils/StorageHelper.kt:111]: nice on, thanks!

**Review by ogkuzmin — APPROVED** [review]
LGTM

**Pull request merged**